### PR TITLE
kcp-dev: add controller-runtime-example repo

### PR DESCRIPTION
**Pull request by stevekuznetsov**
Signed-off-by: Steve Kuznetsov <skuznets@redhat.com>

/cc @ncdc

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
OK, this should be good to go @ncdc

File: ci-operator/config/kcp-dev/controller-runtime-example/kcp-dev-controller-runtime-example-main.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+resources:
+  '*':
+    limits:
+      memory: 4Gi
+    requests:
+      cpu: 100m
+      memory: 200Mi
+tests:
+- as: e2e
+  commands: make test-e2e
```
Comment (ncdc) [inline ci-operator/config/kcp-dev/controller-runtime-example/kcp-dev-controller-runtime-example-main.yaml:14]: odd that this works here directly, vs what we have to do in kcp (tests.step.tests.as)?
```
tests:
- as: lint
  steps:
    test:
    - as: lint
      commands: make lint
```

**Review by ncdc — APPROVED** [review]

File: ci-operator/config/kcp-dev/controller-runtime-example/kcp-dev-controller-runtime-example-main.yaml (hunk 1 above)
Comment (stevekuznetsov) [reply ci-operator/config/kcp-dev/controller-runtime-example/kcp-dev-controller-runtime-example-main.yaml:14]: it's the old-style (what we have here) vs new-style (elsewhere) approach. surprised that the generator gives you this by default

Comment (stevekuznetsov) [reply ci-operator/config/kcp-dev/controller-runtime-example/kcp-dev-controller-runtime-example-main.yaml:14]: https://issues.redhat.com/browse/DPTP-3073

**Review by ncdc — APPROVED** [review]

**Pull request merged**